newlib/libc/tinystdio/stdio.h

```diff
@@ -161,13 +165,25 @@ extern FILE *const stderr;
 #define	_IOLBF	1		/* setvbuf should set line buffered */
 #define	_IONBF	2		/* setvbuf should set unbuffered */
 
-#define fdev_setup_stream(stream, p, g, fl, f)	\
+#ifdef _WANT_FLOCKFILE
+# define fdev_setup_stream(stream, p, g, fl, f)	\
+	do { \
+		(stream)->flags = f; \
+		(stream)->put = p; \
+		(stream)->get = g; \
+		(stream)->flush = fl; \
+		__flockfile_init((stream)); \
```

Comment:
`__flockfile_init` and friends need to be moved to stdio.h for this to work.

Reply:
moved